src/btree/bt_curnext.c

```diff
@@ -732,13 +765,50 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
     WT_SESSION_IMPL *session;
     size_t total_skipped, skipped;
     uint32_t flags;
-    bool newpage, prefix_key_out_of_bounds, restart;
+    int exact;
+    bool newpage, key_out_of_bounds, restart;
 
     cursor = &cbt->iface;
-    prefix_key_out_of_bounds = false;
+    exact = 0;
+    key_out_of_bounds = false;
     session = CUR2S(cbt);
     total_skipped = 0;
 
+    /*
+     * Checks if the cursor is currently positioned and positions it according to the bounds set. If
+     * the cursor is unpositioned, it will be positioned on the lower bound.
+     */
+    if ((&cbt->ref->page) == NULL && F_ISSET(cursor, WT_CURSTD_BOUND_LOWER) &&
+      !(F_ISSET(cursor, WT_CURSTD_BOUND_ENTRY))) {
+        WT_ASSERT(session, WT_DATA_IN_ITEM(&cbt->iface.lower_bound));
+        __wt_cursor_set_raw_key(cursor, &cursor->lower_bound);
```

Comment:
Do we clear raw key on this?

Reply:
Do we need to save the cursor state before clear?

Reply:
When performing a normal cursor next(), we will be repositioning the cursor to a visible record. In doing so, we will be overwriting the key in the end. In the case that we do return WT_NOTFOUND from the cursor->next(), the function will clear the key. I think that we don't need to worry about clearing or restoring the state of the key with bounded cursor.